### DISCUSSION TITLE
Failed when running web crawling script with undetected-chromedriver in headless mode.

### DISCUSSION THREAD

**Discussion by Chen-Ying-Chen** [ultrafunkamsterdam/undetected-chromedriver#1185] [Q&A]
Hi all,

Hope this message finds you well.

I'm writing to ask for help.
I ran the following code for web crawling, succeeded without headless mode , but failed with headless mode and got this error message:

> raise exception_class(message, screen, stacktrace)
> selenium.common.exceptions.WebDriverException: Message: unknown error: cannot connect to chrome at 127.0.0.1:xxxxx
> from chrome not reachable**

Part of my codes:

```
options = uc.ChromeOptions()
options.add_argument('--disable-extensions')
options.add_argument('--disable-gpu')
options.add_argument('--no-sandbox') # Bypass OS security model
options.add_argument('start-maximized')
options.add_argument('disable-infobars')
options.add_argument('--headless=new')
options.add_argument('--disable-dev-shm-usage')
options.add_argument('--remote-debugging-port=9222')
options.add_argument('--disable-browser-side-navigation')
options.add_argument('--disable-features=VizDisplayCompositor')
options.add_argument('--disable-blink-features=AutomationControlled')
driver = uc.Chrome(options=options)
driver.get('https://dune.com/auth/login?next=%2Fqueries%2F32109')
time.sleep(4)
driver.implicitly_wait(30)
.......
```

If you know anything that may fix this problem , please share with me.
Thank you!!

**Comment by Dunkhan** [comment]
I am having the same issue. I tried all the suggestions I could in the discussions here with no change. Does anyone know how to make this library more verbose so that we might get some insight into what is happening?

**Reply by ultrafunkamsterdam** [reply]
Use verbose logging and turn on debug=True

**Comment by va3093** [comment]
You know they have an API right?

https://dune.com/api
